### ISSUE TITLE
"Crashing" During Scene Collection Switch

### ISSUE THREAD

**Issue by buckaroobrogan** [exeldro/obs-source-clone#20]
Hello and Happy New Year!

As of this week, I have been receiving a Source Cleanup Error when trying to switch scenes. I'm pretty sure I have traced the issue to the Source Clone Plugin.
Current installation OBS 30.0.2 (but did attempt with OBS 30.0 with same issues)

It doesn't seem to matter what scene, new or old, empty or complex. I tested this on a portable OBS installation and it came up with the same issue. After testing the issue on safe mode, I was able to connect it to a plugin compatibility issue of some sort. I removed all plugins and then added them one by one until the crash would occur again. Only time I would receive this issue is when Source Clone was installed. I have tried a new installation with the same issues. I cannot recall if the Source Clone Plugin had a recent update, I have been using Andi's StreamUp plugin and I just install updates when I get the notification. I haven't added or removed any new scenes or plugins to make this change occur either. 

I have included the log, but unfortunately I don't see an actual "crash" within it. It doesn't seem like OBS logs in detail when it does this type of error that results in a application shutdown.

TLDR; OBS gives Source Cleanup Error when switching scene collections only when Source Clone is installed and shuts down the application completely.

[2023-12-31 16-54-11 with Source Clone.txt](https://github.com/exeldro/obs-source-clone/files/13802383/2023-12-31.16-54-11.with.Source.Clone.txt)

**Comment by exeldro** [comment]
The log indicates that "DroidCam OBS 3" is the source that is not released and that seems to be a droidcam source. Do you source clone that source? And if so what is different to all other source clones that are released correctly?

**Comment by buckaroobrogan** [comment]
You're 100% correct, it was the source "DroidCam OBS 3"! I have it Source cloned in two different scenes like my other DroidCam sources in OBS, each with their own minimal filters applied. Initially the only obvious differentiating factor was that "DroidCam OBS 3" is the 24/7 plugged in android device, the others 2 are only plugged in for a max of 3 hours. I disconnected the cable to it, removed the source with a new one, restarted OBS with Source Clone installed, and everything is functional again. I didn't even replace the Soure Clones for it, I just reattached them in Properties. Plus, switching scenes collections is a whole lot faster now as well. I don't know how you connected the dots to that specific source, but you are a lifesaver! 

Before I got rid of "DroidCam OBS 3", I tested to see if it had anything else different within it than the other DroidCam sources within the same Scene Collection. What I found was that for some reason, the problem source seemed to be missing the feature of changing image scale within the Properties menu that the others sources have, no matter what device I have in it. I don't know if that's relative to the issue, but definitely strange.

DroidCam had an update December 22nd, which lines up pretty well with the issue as well. When looking at the update, it has this patch listed "Fixed a crash when switching between multiple profiles", though the issue on my end was switching Scene Collections, that patch is decently close.

**Solution:** Deleted the problem DroidCam source and replaced with a new one, unknown why it worked initially without Source Clone plugin installed
